### PR TITLE
Add model_logging module and configuration

### PR DESCRIPTION
**Pull request by adam-thomas**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    extras_require={},
+    zip_safe=False,
+    license='BSD',
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        'Framework :: Django',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: BSD License',
+        'Natural Language :: English',
+        'Programming Language :: Python :: 3.4',
```
Comment (meshy) [inline setup.py:35]: probably worth adding the python 3 only trove classifier

Comment (adam-thomas) [inline setup.py:35]: What's that?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[flake8]
+max-line-length = 90
+max-complexity = 10
+exclude = *migrations*
+statistics = true
+application-import-names = model_logging
+import-order-style = google
+
+[wheel]
+universal = 1
```
Comment (meshy) [inline setup.cfg]: Unless this project intends to support python 2, this is not a universal wheel.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+-e .
+colour-runner==0.0.4
+coverage==4.0.0
+dj-database-url==0.3
+django==1.8.5
+django-pgcrypto-fields==1.0.0
```
Comment (meshy) [inline requirements.txt]: This file should list dev/test requirements, whereas project dependencies should be listed in `setup.py`.

File: setup.py (hunk 1 above)
Comment (meshy) [inline setup.py:35]: it's the one that explicitly says "this project does not support python 2".

`Programming Language :: Python :: 3 :: Only`

See https://pypi.python.org/pypi?%3Aaction=list_classifiers

Comment (meshy) [inline setup.py:35]: probably wants `Programming Language :: Python :: 3` as well.

File: setup.cfg (hunk 2 above)
Comment (adam-thomas) [inline setup.cfg]: oh whoops, I copied that from elsewhere

File: requirements.txt (hunk 3 above)
Comment (adam-thomas) [inline requirements.txt]: ok :)

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+djangorestframework==3.2.4
+factory_boy==2.5.2
+flake8-import-order==0.5.1
+incuna-test-utils==6.3.1
+psycopg2==2.6.1
+simplejson==3.8.0
+wheel==0.24.0
```
Comment (LilyFirefly) [inline requirements.txt]: Should we use the `~=` compatible release operator here?

File: setup.py (hunk 1 above)
Comment (LilyFirefly) [inline setup.py:35]: Should we be officially supporting python 3.5 too?

File: requirements.txt (hunk 4 above)
Comment (adam-thomas) [inline requirements.txt]: No?

**Comment by adam-thomas** [comment]
@incuna/backend OK I think this is ready to go now!  Review please? :)

File: setup.py (hunk 1 above)
Comment (adam-thomas) [inline setup.py:35]: Is it out yet?  I don't think any of our other packages are right now.

Comment (LilyFirefly) [inline setup.py:35]: Yes - it was released about 3 weeks ago.

File: model_logging/serializers.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+class JSONField(Field):
+    def to_representation(self, obj):
+        return json.loads(obj)
+
+
+class LogEntrySerializer(ModelSerializer):
+    data = JSONField()
+    creator = CharField(source='creator.email')
+    operation_label = CharField(source='get_operation_display', read_only=True)
```
Comment (kevinetienne) [inline model_logging/serializers.py:20]: should this be a `ChoiceField`?

**Comment by kevinetienne** [comment]
Looks good to me, maybe we will need a `readme.md` in the future?

**Comment by meshy** [comment]
> ...we will need a `readme.md`

...and a changelog!

File: model_logging/serializers.py (hunk 5 above)
Comment (adam-thomas) [inline model_logging/serializers.py:20]: Nah, it's read only :)

**Comment by adam-thomas** [comment]
oh yes! I shall add those.

File: model_logging/serializers.py (hunk 5 above)
Comment (kevinetienne) [inline model_logging/serializers.py:20]: thanks yeah forgot about that

**Comment by adam-thomas** [comment]
README and changelog added :)

File: README.md [hunk 6]
```diff
@@ -1 +1,96 @@
...
+
+This library uses [`django-pgcrypto-fields`](https://github.com/incuna/django-pgcrypto-fields),
+which means you will need PGP public and private keys.  See the `settings.configure()`
+block in `model_logging/tests/run.py` for an (extremely insecure) example.
+
+## Usage
+
+#### Low-level use
+
+```
```
Comment (kevinetienne) [inline README.md]: can we enable syntax highlighting by adding `python` after the backticks please?

Comment (adam-thomas) [inline README.md]: Oh, does that work in readmes?  Sure!

**Comment by adam-thomas** [comment]
Thanks :D

**Pull request merged**